### PR TITLE
[release/CI] Persist release test metrics in DB

### PR DESCRIPTION
**Pull request by Yard1**
## Why are these changes needed?

This PR adds a system to our release test infrastructure to obtain Prometheus metrics (eg. cpu utilization, same as displayed in Grafana) from the Anyscale head node and persist them in the databricks delta lake for later analysis.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ray_release/glue.py [hunk 1]
```diff
@@ -289,6 +291,14 @@ def run_release_test(
             logger.exception(e)
             command_results = {}
 
+        try:
+            command_runner.save_metrics(start_time_unix)
+            metrics = command_runner.fetch_metrics()
+        except Exception as e:
+            logger.error("Could not fetch metrics for test command")
+            logger.exception(e)
```
Comment (krfricke) [inline release/ray_release/glue.py:299]: `logger.exception` is just `logger.error` with printing the traceback, so we should probably do this for better readability

```suggestion
            logger.exception(f"Could not fetch metrics for test command: {e}")
```

File: release/ray_release/command_runner/job_runner.py [hunk 2]
```diff
@@ -59,6 +59,15 @@ def prepare_remote_env(self):
             os.unlink("wait_cluster.py")
         os.link(wait_script, "wait_cluster.py")
 
+        # Copy prometheus metrics script to working dir
+        metrics_script = os.path.join(
+            os.path.dirname(__file__), "_prometheus_metrics.py"
+        )
+        # Copy wait script to working dir
```
Comment (krfricke) [inline release/ray_release/command_runner/job_runner.py:66]: nit
```suggestion
        # Copy prometheus metrics script to working dir
```

**Review by krfricke — COMMENTED** [review]
Release infra code looks good to me. Have we clarified how we want to store data/manage data lifecycle?

**Comment by Yard1** [comment]
@krfricke Yes, we decided to just save the data in the release test table as it shouldn't be an issue on the cost side to keep it there. We can set up an automated query to remove data older than N-months every month later.

**Comment by Yard1** [comment]
Tested manually with all 3 different release test types we have (client, job, sdk) and metrics were persisted successfully in each case.

**Review by krfricke — APPROVED** [review]
Great, thanks!

**Comment by rkooo567** [comment]
@Yard1 CAn you tell me how to query this? This is a nice feature... We can also plot some of them (e.g., memory growth) to  the Preset

**Comment by rkooo567** [comment]
cc @scv119 @jjyao

**Comment by Yard1** [comment]
@rkooo567 `SELECT * FROM hive_metastore.oss_ci.ray_release_test_result where cpu_count IS NOT NULL` in databricks

The data is raw prometheus output. You should be able to just put it directly into grafana to get the same output as you would in Anyscale. It's separated by node and the interval between data points is 15 seconds. I was thinking about doing some offline analysis with python after a month or two of data has been collected.

**Comment by jjyao** [comment]
Actually for the schema, is `cpu_count` under `prometheus_metrics`? I looked at the table schema, seems we added 10+ top level fields?

**Pull request merged**